### DISCUSSION TITLE
Merging 2023 Q1 changes

### DISCUSSION THREAD

**Discussion by deadwood2** [aros-development-team/AROS#603] [General]
Hi,

This is a thread for merging changes from Q1 2023. I start with changes which should post low risk. The PR is now submitted.

**Comment by deadwood2** [comment]
Apart from that first batch of changes, there should be 3 more: mui, kernel, c library

**Comment by Kalamatee** [comment]
On Mon, 3 Apr 2023 at 18:59, deadw00d ***@***.***> wrote:

> Apart from that first batch of changes, there should be 3 more: mui,
> kernel, c library
>
Cool - I look forward to seeing them =)
I have merged the first changes now and am building to make sure all is
well.

Regarding the last one -:
// Wait for reset to complete (spec is 50ms, FreeBSD source suggests 200ms,
but
// we compromise to help USB volumes become available in time to be chosen
as
// the boot device)
uhwDelayMS(125, unit);

might it not be better to just iterate 4 times waiting 50ms, and check each
time if it is ready?

**Reply by deadwood2** [reply]
I don't remember, it was a long time ago. If you want to change this, keep in mind:

1) The original problem was that switch from OHCI to EHCI was not working under VirtualBox (change from 50 -> 200)
2) The additional problem addressed in this commit was that 200 was too long for some hardware and was causing going for ATA instead of USB (commit message says so)

**Comment by Kalamatee** [comment]
On Mon, 3 Apr 2023 at 19:10, deadw00d ***@***.***> wrote:

> I don't remember, it was a long time ago. If you want to change this, keep
> in mind:
>
>    1. The original problem was that switch from OHCI to EHCI was not
>    working under VirtualBox (change from 50 -> 200)
>    2. The additional problem addressed in this commit was that 200 was
>    too long for some hardware and was causing going for ATA instead of USB
>    (commit message says so)
>
>
yep got it - so if we do 50ms waits and check until its ready it should
satisfy both - (I will look into it more when working on usb..)

Im not sure if you have access to codacy  - but I get it to check the code
just incase there are obvious issues. Most of this looks pretty benign but
I think there may be a few actual things that need looked at? AROS - Codacy
- PullRequest
<https://app.codacy.com/gh/aros-development-team/AROS/pullRequest?prid=11488316&bid=34645003>

**Reply by deadwood2** [reply]
Yes, I saw them. The critical ones happens in Neil's drivers. I'll forward this result to him.

**Comment by deadwood2** [comment]
Note, I also submitted a 2-commit PR for contrib

**Comment by Kalamatee** [comment]
On Tue, 4 Apr 2023 at 11:33, deadw00d ***@***.***> wrote:

> Note, I also submitted a 2-commit PR for contrib
>

Thanks again - I have merged the contrib and MUI ones, but will need to
think on the fpos/ino changes (I think they are meant to be decided by the
compiler flags, but need to look into it ... my heads too full of xhci
stuff just now)

**Reply by deadwood2** [reply]
Sure, that's fine. With fpos/ino I was trying to match what is already there in headers. Anyhow if you have any further questions, let me know.

**Comment by Kalamatee** [comment]
On Tue, 4 Apr 2023 at 12:01, Kalamatee ***@***.***> wrote:

> . my heads too full of xhci stuff just now
>

I have enough boilerplate/stubs in place in the pciusb driver to have it
find the usb 3 controller, and expose some details so far ... but am just
starting to put in the actual chipset control code

[image: image.png][image: image.png]

**Comment by deadwood2** [comment]
FYI, I submitted all but 1 PRs. The remaining one is on the kernel changes (~6 commits). I will submit it once I get the kernel with pulled in changes up and running.

**Comment by Kalamatee** [comment]
On Tue, 4 Apr 2023 at 12:30, deadw00d ***@***.***> wrote:

> FYI, I submitted all but 1 PRs. The remaining one is on the kernel changes
> (~6 commits). I will submit it once I get the kernel with pulled in changes
> up and running.
>

Perfect, thanks.

**Comment by deadwood2** [comment]
I submitted the last part of changes. I'm aware of one regression in muimaster.library that is caused by one of previous PRs and I will submit a fix for it as well still in this cycle.

**Comment by Kalamatee** [comment]
Thanks Deadwood - I have merged the changes.I will probably need to email
you about the acpi related patches (shouldnt strictly be necessary?) but
otherwise it all looks good from this side.

**Comment by deadwood2** [comment]
FYI, I pushed the final PR for this synchronization work.
